vendorconsent/tcf2/metadata.go

```diff
@@ -118,29 +129,33 @@ func (c ConsentMetadata) ConsentLanguage() string {
 	return string([]byte{leftChar + 65, rightChar + 65}) // Unicode A-Z is 65-90
 }
 
+// VendorListVersion returns the vendor list version stored in bits 120 to 131
 func (c ConsentMetadata) VendorListVersion() uint16 {
 	// The vendor list version is stored in bits 120 - 131
 	rightByte := ((c.data[16] & 0xf0) >> 4) | ((c.data[15] & 0x0f) << 4)
 	leftByte := c.data[15] >> 4
 	return binary.BigEndian.Uint16([]byte{leftByte, rightByte})
 }
 
+// MaxVendorID returns the maximum value for vendor identifier in bits 213 to 228
 func (c ConsentMetadata) MaxVendorID() uint16 {
 	// The max vendor ID is stored in bits 213 - 228 [00000xxx xxxxxxxx xxxxx000]
 	leftByte := ((c.data[26] & 0x07) << 5) | ((c.data[27] & 0xf8) >> 3)
 	rightByte := ((c.data[27] & 0x07) << 5) | ((c.data[28] & 0xf8) >> 3)
 	return binary.BigEndian.Uint16([]byte{leftByte, rightByte})
 }
 
+// PurposeAllowed returns if the given purpose (1 to 24 max) is enabled, info stored in bits 152 to 175
 func (c ConsentMetadata) PurposeAllowed(id consentconstants.Purpose) bool {
 	// Purposes are stored in bits 152 - 175. The interface contract only defines behavior for ints in the range [1, 24]...
 	// so in the valid range, this won't even overflow a uint8.
 	if id > 24 {
-		id = 24
+		id = 24 // tiago: why this behavior is different than PurposeLITransparency? should we check for 0 and negative too?
```

Comment:
This is inconsistent behavior IMO. If the id is greater than the range, I think we should return false immediately. Right now, it's returning the highest purpose possible.. which is, strange.

No need to check for negative since the id is an unsigned int.

I'm not sure if we should add checks for != 0, @hhhjort?

Reply:
I am not sure how it became inconsistent. We should probably make it consistent. The >24 check is more important than != 0, as the bit associated with 0 definitely exists, but we may overrun the consent string for an ID high enough. Still not a bad idea to include it.

Also a good point it may be better to just return a false for out of bounds, it is a bit more predictable.